src/api_manager/method_impl.h

```diff
@@ -154,6 +154,16 @@ class MethodInfoImpl : public MethodInfo {
 
   void ProcessSystemQueryParameterNames();
 
+  bool escape_binding() const override {
+    // Variable bindings normaly are used for grpc transcoding.
+    // Their values should be un-escaped.
+    // But for CONSTANT_ADDRESS bakend path translation,
+    // they are used to re-format the URL path: from path segments
+    // to query parameters. They should remain escaped.
+    return backend_path_translation() ==
```

Comment:
Not sure about this. 

Say for APPEND_PATH_TO_ADDRESS, if
Request path: /api/company/%20widgetworks/user/johndoe?timezone=%20EST

the translated path should also keep the escape character, like:
https://example.appspot.com/api/company/%20widgetworks/user/johndoe?timezone=%20EST

right?

Reply:
Yes.  actually it should do that for both path_translation, not just for APPEND_PATH_TO_ADDRESS.

Reply:
But CONSTANT_ADDRESS is using variable_bindings.  By default, variable_bindings un-escape the binding values.  It is OK for transcoding.  but it is not OK for backend routing.